open_academy/models/open_academy_session.py

```diff
@@ -1,14 +1,53 @@
-from odoo import models, fields
+from odoo import models, fields, api
+from odoo.exceptions import ValidationError
 
 
 class OpenAcademySession(models.Model):
     _name = "open.academy.session"
     _description = "model of session on open academy"
 
     name = fields.Char(required=True, size=50)
-    duration = fields.Text()
-    start_date = fields.Datetime()
-    number_seats = fields.Integer()
+    duration = fields.Integer()
+    active = fields.Boolean(default=True)
+    start_date = fields.Datetime(default=fields.Date.context_today)
+    end_date = fields.Datetime()
+    number_seats = fields.Integer(required=True, default=1)
+    number_seats_taken = fields.Integer()
     instructor_id = fields.Many2one("res.partner", "instructor")
     course_id = fields.Many2one("open.academy.course", required=True)
     attendees_ids = fields.Many2many("res.partner", "session_res_partner_rel")
+    available_seats = fields.Integer(compute="_compute_percentage_available_seats")
+
+    @api.depends("number_seats", "number_seats_taken")
+    def _compute_percentage_available_seats(self):
+        for records in self:
+            available = records.number_seats - records.number_seats_taken
+            if records.number_seats_taken == 0:
+                records.available_seats = 100
+                return
+            records.available_seats = (available / records.number_seats) * 100
+
+    @api.onchange("number_seats", "number_seats_taken", "attendees_ids")
+    def _onchange_number_seats(self):
+        taken = len(self.attendees_ids)
+        self.number_seats_taken = taken
```

Comment:
If you do the computed for this field you don't need this change. Please add the computed method for number_seats_taken field.